xgcm/test/test_grid.py

```diff
@@ -631,6 +635,7 @@ def test_grid_ops(all_datasets):
                     da_cumsum = grid.cumsum(ds[varname], axis_name, boundary=boundary)
                     da_cumsum_ax = axis.cumsum(ds[varname], boundary=boundary)
                     assert da_cumsum.equals(da_cumsum_ax)
+                # test diff keep_coords?
```

Comment:
I personally would write an additional test function that tests the behavior of keep_attrs on a high level, instead of testing this in all the other tests. That way we can be sure that the existing behavior is not modified.

Reply:
could you please be a bit more specific about this last point, I am not following sorry

Reply:
I think looking at the [tests for the metrics capability](https://github.com/xgcm/xgcm/blob/master/xgcm/test/test_metric_ops_multi_axis.py), could be helpful to get you started. I was talking about something along the lines of:
```
from xgcm.test.datasets import datasets_grid_metric

@pytest.mark.parametrize("funcname", ["diff", "interp", "min", "max", "integrate", "average", "cumint"]) # this should cover every operation that we have available
def test_keep_coords(all_datasets,funcname):
    ds, coords, metrics = datasets_grid_metric("B")
    grid = Grid(ds, coords=coords, metrics=metrics, periodic=periodic)
    func = getattr(grid, funcname) #grab the function from the grid object insead of typing out every instance of grid.diff, grid.interp....
    for axis_name in grid.axes.keys():
        result = func(ds.var, axis_name)
        result_w_coords = func(ds.var, axis_name, keep_coords=True)
        # some logic that asserts that `results` has no coords except the dimensions, and `results_w_coords` does in fact have all of the desired coordinates
```
The test datasets should have plenty of non-dim coordinates at different positions (the grid metrics).

This will make sure that we consistently can pass the argument to all available functions and that indeed the coordinates are preserved when we pass it. Let me know if this makes sense (the last message was written too late at night, apologies).

Reply:
thanks a bunch for this enlightening guidance.
a81166c implements a first test